cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/template/RemoteTemplateStorage.java

```diff
@@ -0,0 +1,331 @@
+            this.throwException(response, () -> ProtocolBuffer.readAll(chunkedResponse.getInputStream()).readThrowable());
+            return chunkedResponse.getInputStream();
+        });
+    }
+
+    @Override
+    public @NotNull ITask<FileInfo> getFileInfoAsync(@NotNull ServiceTemplate template, @NotNull String path) {
+        return this.executeDriverAPIMethod(
+                DriverAPIRequestType.GET_FILE_INFO,
+                buffer -> this.writeDefaults(buffer, template).writeString(path),
+                packet -> packet.getBuffer().readOptionalObject(FileInfo.class)
+        );
+    }
+
+    @Override
+    public @NotNull ITask<FileInfo[]> listFilesAsync(@NotNull ServiceTemplate template, @NotNull String dir, boolean deep) {
+        return this.executeDriverAPIMethod(
+                DriverAPIRequestType.LIST_FILES,
+                buffer -> this.writeDefaults(buffer, template).writeString(dir).writeBoolean(deep),
+                packet -> packet.getBuffer().readBoolean() ? packet.getBuffer().readObjectArray(FileInfo.class) : null
+        );
+    }
+
+    @Override
+    public @NotNull ITask<Collection<ServiceTemplate>> getTemplatesAsync() {
+        return this.executeDriverAPIMethod(
+                DriverAPIRequestType.GET_TEMPLATES,
+                packet -> packet.getBuffer().readObjectCollection(ServiceTemplate.class)
+        );
+    }
+
+    @Override
+    public @NotNull ITask<Void> closeAsync() {
+        return this.executeVoidDriverAPIMethod(
+                DriverAPIRequestType.CLOSE_STORAGE,
+                this::writeDefaults
+        );
+    }
+
+    @CanIgnoreReturnValue
+    private ProtocolBuffer writeDefaults(ProtocolBuffer buffer, ServiceTemplate template) {
+        return buffer.writeObject(template);
+    }
+
+    @CanIgnoreReturnValue
+    private ProtocolBuffer writeDefaults(ProtocolBuffer buffer) {
+        return buffer.writeString(this.name);
+    }
+
+    @CanIgnoreReturnValue
+    private IPacket readDefaults(IPacket packet) throws IOException {
+        TemplateStorageResponse response = packet.getBuffer().readEnumConstant(TemplateStorageResponse.class);
+        this.throwException(response, packet.getBuffer());
+        return packet;
+    }
+
+    private boolean readDefaultBooleanResponse(IPacket packet) throws IOException {
+        TemplateStorageResponse response = packet.getBuffer().readEnumConstant(TemplateStorageResponse.class);
+        this.throwException(response, packet.getBuffer());
+        return response == TemplateStorageResponse.SUCCESS;
+    }
+
+    private void throwException(TemplateStorageResponse response, ProtocolBuffer buffer) throws IOException {
+        this.throwException(response, buffer::readThrowable);
+    }
+
+    private void throwException(TemplateStorageResponse response, Supplier<Throwable> throwableSupplier) throws IOException {
+        switch (response) {
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=6033851)